src/ray/raylet/task_dependency_manager.cc

```diff
@@ -297,6 +298,44 @@ void TaskDependencyManager::TaskCanceled(const TaskID &task_id) {
   }
 }
 
+void TaskDependencyManager::CleanupForDriver(const std::unordered_set<TaskID> &task_ids) {
+  for (auto it = task_dependencies_.begin(); it != task_dependencies_.end();) {
+    if (task_ids.find(it->first) != task_ids.end()) {
+      it = task_dependencies_.erase(it);
+    } else {
+      it++;
+    }
+  }
+
+  for (auto it = required_tasks_.begin(); it != required_tasks_.end();) {
+    if (task_ids.find(it->first) != task_ids.end()) {
+      it = required_tasks_.erase(it);
+    } else {
+      it++;
+    }
+  }
+
+  for (auto it = pending_tasks_.begin(); it != pending_tasks_.end();) {
+    if (task_ids.find(it->first) != task_ids.end()) {
+      it = pending_tasks_.erase(it);
+    } else {
+      it++;
+    }
+  }
```

Comment:
Create a function or lambda to for the three similar code

Reply:
good point.